### ISSUE TITLE
Ubiquity changes not imported after store reload

### ISSUE THREAD

**Issue by henneonrails** [lhunath/UbiquityStoreManager#2]
I am very happy with your code. But I have one problem: when data changes on one device the notification is thrown. But how to populate? I have registered the `UbiquityManagedStoreDidImportChangesNotification` also to `reloadFetchedResults`.

I think that I always get the old `managedObjectContext` from `managedObjectContextForUbiquityChangesInManager` 

To be clear, when I start the app again the new data is shown. Syncing does work, I have no errors. The logs are fine. But I cannot populate.

**Comment by henneonrails** [comment]
`self.managedObjectContext.stalenessInterval = 0;`solved the problem.

**Comment by lhunath** [comment]
The `managedObjectContextForUbiquityChangesInManager` method is one you're supposed to implement and return your application's MOC from.  You don't call it, and you don't get anything from it.  USM will then import the ubiquity changes into the MOC you return from this method, allowing your application to immediately see the changes.

Setting `stalenessInterval` to zero is not recommended as it defeats your MOC's caching.

**Comment by henneonrails** [comment]
I understand what you described, but in my project this does not work like you describe it. The changes are not imported. Only after a "restart" of the app. Setting `stalenessInterval` to 0 solves the problem. I understand that this is not the right way to do it, but I don't know what else to do.

**Comment by lhunath** [comment]
What do you do to check for updated entities?  Are you fetching from the same MOC instance as the one that you're giving to USM?

**Comment by henneonrails** [comment]
I listen to the `UbiquityManagedStoreDidImportChangesNotification` and call the same method as the `UbiquityManagedStoreDidChangeNotification` in the sample code. Here is the code:

``` objective-c
- (void)reloadFetchedResults:(NSNotification*)note {

    // Refetch the data
    self.fetchedResultsController = nil;
    [self fetchedResultsController];

    if (note) {
        [self.tableView reloadData];

        // STEP 5b - Display current state of the UbiquityStoreManager
        BOOL enabled = [[AppDelegate appDelegate] ubiquityStoreManager].cloudEnabled;
        [iCloudSwitch setOn:enabled animated:YES];
    }
}
```

**Comment by Tw0bit** [comment]
@lhunath / @henneonrails ,

It appears the issue is with commit https://github.com/lhunath/UbiquityStoreManager/commit/2586f75779570ae43442ed52a82ac0acb55caa01

If you use any previous commit, the notifications work.. It appears that line 311 is missing the IF STATEMENT, which was there in the previous releases:

if ([self.persistentStoreCoordinator.persistentStores count])
        // We couldn't remove all the stores, make a new PSC instead.
    [self resetPersistentStoreCoordinator];

**Comment by lhunath** [comment]
The store shouldn't get cleared when importing ubiquity changes.

Moreover, the fact that the PSC changes when clearing the store shouldn't cause your MOCs to be outdated unless they're still hooked up to the old PSC.  To that end, please make sure your `-ubiquityStoreManager:willLoadStoreIsCloud:` implementation unsets your application's MOCs consistently and your `-ubiquityStoreManager:didLoadStoreForCoordinator:isCloud:` creates new MOCs for the given PSC each time.

**Comment by Tw0bit** [comment]
Hi lhunath,

We have been using the original Library for 10months, and migrated to your version recently (still in development). And have not had this issue until your last commit... 

In our AppDelegate, we have setup the delegates to the ubiquity store managers as per the readme/sample documentation.

Without that extra IF in the ClearStore function, the registered notifications are not triggered.

The code for the methods you mentioned are:

```
- (NSManagedObjectContext *)managedObjectContextForUbiquityChangesInManager:(UbiquityStoreManager *)manager
{
    return self.managedObjectContext;
}

- (void)ubiquityStoreManager:(UbiquityStoreManager *)manager willLoadStoreIsCloud:(BOOL)isCloudStore
{    
    _managedObjectContext = nil;
    self.isReady = FALSE;

    if (![self.handleLoadingStoreAlert isVisible]) // Ensure only one popup is presented (else "hangs" app)
    {
        dispatch_async(dispatch_get_main_queue(), ^{
            self.handleLoadingStoreAlert = [[UIAlertView alloc] initWithTitle:@"Loading Datastore" message:
                                            @"\n\n\n\nPlease wait, setting up data..."
                                                                     delegate:nil
                                                            cancelButtonTitle:nil otherButtonTitles:nil];
            UIActivityIndicatorView *activityIndicator = [[UIActivityIndicatorView alloc]
                                                          initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
            activityIndicator.center = CGPointMake( 142, 90 );
            [activityIndicator startAnimating];
            [self.handleLoadingStoreAlert addSubview:activityIndicator];
            [self.handleLoadingStoreAlert show];
        });
    }
}

- (void)ubiquityStoreManager:(UbiquityStoreManager *)manager failedLoadingStoreWithCause:(UbiquityStoreErrorCause)cause context:(id)context wasCloud:(BOOL)wasCloudStore {
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.handleLoadingStoreAlert dismissWithClickedButtonIndex:0 animated:YES];

        if (!wasCloudStore && ![self.handleLocalStoreAlert isVisible]) {
            self.handleLocalStoreAlert = [[UIAlertView alloc] initWithTitle:@"Local Store Problem" message:
                                          @"Your datastore got corrupted and needs to be recreated."
                                                                   delegate:self
                                                          cancelButtonTitle:nil otherButtonTitles:@"Start Over", nil];
            [self.handleLocalStoreAlert show];
        }
    });
}

- (void)ubiquityStoreManager:(UbiquityStoreManager *)manager didLoadStoreForCoordinator:(NSPersistentStoreCoordinator *)coordinator isCloud:(BOOL)isCloudStore {

    NSManagedObjectContext *moc = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSMainQueueConcurrencyType];
    [moc setPersistentStoreCoordinator:coordinator];

    _managedObjectContext = moc;
    _managedObjectContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy;

    dispatch_async(dispatch_get_main_queue(), ^{
        [self.handleCloudContentAlert dismissWithClickedButtonIndex:[self.handleCloudContentAlert firstOtherButtonIndex] + 9
                                                           animated:YES];
        [self.handleCloudContentWarningAlert dismissWithClickedButtonIndex:[self.handleCloudContentWarningAlert firstOtherButtonIndex] + 9
                                                                  animated:YES];

        [self.handleLoadingStoreAlert dismissWithClickedButtonIndex:0 animated:YES];
    });
}
```

**Comment by lhunath** [comment]
That looks right.  Can you log the following:

In `managedObjectContextForUbiquityChangesInManager:`, output the MOC instance and the PSC instance associated with it.

In `willLoadStoreIsCloud`, log the MOC instance before you unset it and the PSC associated with it.

In `didLoadStoreForCoordinator`, log the old MOC instance and the PSC associated with it before setting the new one, and log the new MOC and the PSC associated with that one after setting it.

Then, cause some changes to get imported and see whether the changes become visible after import, and see what the log output is if not.

Let's get some visibility into what exactly is happening.

**Comment by Tw0bit** [comment]
App Design: TabBar Controller (Storyboard targeted to iOS6) with 5 Views. Each View is a CoreData Table using the FetchResultsController implementation from Apple. All views in viewDidLoad monitor the two nominated Notification as follows (and reloads the data):

```
// Observe the app delegate telling us when it's finished asynchronously setting up the persistent store
    [[NSNotificationCenter defaultCenter] addObserver: self
                                             selector: @selector(reloadFetchedResults:)
                                                 name: UbiquityManagedStoreDidChangeNotification
                                               object: [[myAppDelegate appDelegate] ubiquityStoreManager]];

    // Observe the app delegate telling us when a merge occurs
    [[NSNotificationCenter defaultCenter] addObserver: self
                                             selector: @selector(reloadFetchedResults:)
                                                 name: UbiquityManagedStoreDidImportChangesNotification
                                               object: [[myAppDelegate appDelegate] ubiquityStoreManager]];
```

I have added the following NSLogs to the three functions you requested:

```
NSLog(@"functionNameHere: MOC - %@", self.managedObjectContext);
    NSLog(@"functionNameHere: PSC - %@", self.managedObjectContext.persistentStoreCoordinator);
```

My Findings when running the LATEST Commit:
iDevice: Launch with iCloud Enabled:

2013-05-09 00:15:56:063 AppNameHere[8340:2311] UbiquityStoreManager: (Re)loading store...
2013-05-09 00:15:56:070 AppNameHere[8340:6403] UbiquityStoreManager: Will load cloud store: A8E08CCD-40B5-40F5-A8FA-C336A49D22C3 (definite).
2013-05-09 00:15:56:082 AppNameHere[8340:6403] UbiquityStoreManager: Clearing stores...
2013-05-09 00:15:56:086 AppNameHere[8340:6403] willLoadStoreIsCloud: MOC - (null)
2013-05-09 00:15:56:089 AppNameHere[8340:6403] willLoadStoreIsCloud: PSC - (null)

2013-05-09 00:15:57:061 AppNameHere[8340:6403] UbiquityStoreManager: Cloud enabled and successfully loaded cloud store.
2013-05-09 00:15:57:064 AppNameHere[8340:2311] didLoadStoreForCoordinator: MOC (OLD) - (null)
2013-05-09 00:15:57:070 AppNameHere[8340:2311] didLoadStoreForCoordinator: PSC (OLD) - (null)
2013-05-09 00:15:57:073 AppNameHere[8340:2311] didLoadStoreForCoordinator: MOC - (NEW) <NSManagedObjectContext: 0x1e5f52f0>
2013-05-09 00:15:57:076 ve4dInsurApp[8340:2311] didLoadStoreForCoordinator: PSC - (NEW) <NSPersistentStoreCoordinator: 0x1f09e0a0>

iDevice: When data sent from second device:

No further calls to functions: managedObjectContextForUbiquityChangesInManager, willLoadStoreIsCloud, didLoadStoreForCoordinator (and no UbiquityStoreManager appears in log)
No Data updates within app., Console shows the Apple Log information for the transactions - however the manager code does not act on it:

```
2013-05-09 00:22:19.686 AppNameHere[8363:907] -[PFUbiquityStoreSaveSnapshot createUbiquityDictionary:withStoreExportContext:error:](499): CoreData: Ubiquity:  Got final value for relationship: ddPaymentDetails
(
)
Object: <DeviceDetails: 0x20b93c20> (entity: DeviceDetails; id: 0x20af3f40 <x-coredata://46DFFF0D-38FD-4F4C-B7A1-0B586696744D/DeviceDetails/p1025> ; data: {
    ddDeviceTitle = "iPhone";
})
2013-05-09 00:22:19.688 AppNameHere[8363:907] -[_PFUbiquityRecordsExporter createDictionaryForObjectsInSaveNotification:forTransactionOfType:withExportContext:andSaveSnapshot:](543): CoreData: Ubiquity:  <_PFUbiquityRecordsExporter: 0x1fd70a40>
    localPeerID: mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15
    ubiquityRootLocation: <PFUbiquityLocation: 0x1fd71600>: /private/var/mobile/Library/Mobile Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3
    tempMoveTimer: (null)
    lastTransactionDate: 2013-05-08 14:22:19 +0000
 exporting deleted objects: (null)
2013-05-09 00:22:19.690 AppNameHere[8363:907] +[PFUbiquityLocation createUbiquityLocationForPath:withUbiquityRootPath:](549): CoreData: Ubiquity:  Path: /private/var/mobile/Library/Mobile Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3 is a ubiquity root url.
Matches: /private/var/mobile/Library/Mobile Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3
2013-05-09 00:22:19.717 AppNameHere[8363:907] -[_PFUbiquityRecordsExporter managedObjectContextDidSave:](429): CoreData: Ubiquity:  <_PFUbiquityRecordsExporter: 0x1fd70a40>
    localPeerID: mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15
    ubiquityRootLocation: <PFUbiquityLocation: 0x1fd71600>: /private/var/mobile/Library/Mobile Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3
    tempMoveTimer: <__NSCFTimer: 0x20a7b930>
    lastTransactionDate: 2013-05-08 14:22:19 +0000
 Successfully wrote transaction log: <PFUbiquityTransactionLog: 0x20a7b9e0>
    transactionLogLocation: <PFUbiquityLocation: 0x20a7bc30>: /private/var/mobile/Library/Mobile Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3/mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15/UbiquityStore/NoEy7Wrw5dPf_BtfqUtgV2I14KZ0Mb2fHX_YLcXRLSM=/B1C13080-3022-48E6-87B5-081A6DF7C010.1.cdt
    transactionNumber: 2
2013-05-09 00:22:19.719 AppNameHere[8363:907] -[_PFUbiquityRecordsExporter managedObjectContextDidSave:](437): CoreData: Ubiquity:  About to save stack: <_PFUbiquityStack: 0x20b97cd0>: 
    localPeerID: mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15
    ubiquityRootURL: <PFUbiquityLocation: 0x208d3030>: /private/var/mobile/Library/Mobile Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3
    metadataContainerURL: <PFUbiquityLocation: 0x208ccd80>: /private/var/mobile/Library/Mobile Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3/.cdmetadata/metadata.nosync/mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15/metadata.store
2013-05-09 00:22:19.723 AppNameHere[8363:907] -[_PFUbiquityRecordsExporter managedObjectContextDidSave:](438): CoreData: Ubiquity:  Stack Changes:
Inserted: {(
    <PFUbiquityTransactionEntry: 0x20be2ca0> (entity: PFUbiquityTransactionEntry; id: 0x20ba4b30 <x-coredata:///PFUbiquityTransactionEntry/tF1CBC536-F13D-4EAE-8126-DE2C2518977E11> ; data: {
    actingPeer = "0x20bd8050 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityPeer/p2>";
    globalIDStr = "UbiquityStore:InsurerDetails:p1025:mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15";
    knowledgeVectorString = "mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15:2:mobile.7F77633C-F13B-518E-9A63-8D5EB212D8AD:0";
    localIDStr = "x-coredata://46DFFF0D-38FD-4F4C-B7A1-0B586696744D/InsurerDetails/p1025";
    storeMetadata = "0x208e0b90 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityStoreMetadata/p1>";
    transactionDate = "2013-05-08 14:22:19 +0000";
    transactionLogFilename = "B1C13080-3022-48E6-87B5-081A6DF7C010.1.cdt";
    transactionNumber = 2;
    transactionTypeNum = 0;
}),
    <PFUbiquityTransactionEntry: 0x20af64b0> (entity: PFUbiquityTransactionEntry; id: 0x20a7d6f0 <x-coredata:///PFUbiquityTransactionEntry/tF1CBC536-F13D-4EAE-8126-DE2C2518977E12> ; data: {
    actingPeer = "0x20bd8050 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityPeer/p2>";
    globalIDStr = "UbiquityStore:DeviceDetails:p1025:mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15";
    knowledgeVectorString = "mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15:2:mobile.7F77633C-F13B-518E-9A63-8D5EB212D8AD:0";
    localIDStr = "x-coredata://46DFFF0D-38FD-4F4C-B7A1-0B586696744D/DeviceDetails/p1025";
    storeMetadata = "0x208e0b90 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityStoreMetadata/p1>";
    transactionDate = "2013-05-08 14:22:19 +0000";
    transactionLogFilename = "B1C13080-3022-48E6-87B5-081A6DF7C010.1.cdt";
    transactionNumber = 2;
    transactionTypeNum = 1;
})
)}
Updated: {(
    <PFUbiquityStoreMetadata: 0x20b9d1f0> (entity: PFUbiquityStoreMetadata; id: 0x208e0b90 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityStoreMetadata/p1> ; data: {
    modelVersionHashString = "NoEy7Wrw5dPf_BtfqUtgV2I14KZ0Mb2fHX_YLcXRLSM=";
    peerStates = "<relationship fault: 0x20ba2510 'peerStates'>";
    primaryKeyRanges = "<relationship fault: 0x20ba2240 'primaryKeyRanges'>";
    storeIdentifier = "46DFFF0D-38FD-4F4C-B7A1-0B586696744D";
    storeOptionsBinaryPlistData = <62706c69 73743030 d4010203 04050831 32542474 6f705824 6f626a65 63747358 24766572 73696f6e 59246172 63686976 6572>;
    storeType = SQLite;
    storeURLString = "file://localhost/private/var/mobile/Library/Mobile%20Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Database.nosync/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3.sqlite";
    transactionEntries =     (
        "0x20bb2340 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p2>",
        "0x20bd4e20 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p34>",
        "0x20ba5380 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p12>",
        "0x20ba4b30 <x-coredata:///PFUbiquityTransactionEntry/tF1CBC536-F13D-4EAE-8126-DE2C2518977E11>",
        "0x20bd4a80 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p29>",
        "0x20ba5360 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p11>",
        "0x20bd4a70 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p28>",
        "0x20bde1a0 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p10>",
        "0x20bd4a60 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p27>",
        "0x20b969d0 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p9>",
        "(...and 26 more...)"
    );
    ubiquityName = UbiquityStore;
    ubiquityRelativePath = nil;
    ubiquityRootURLString = "file://localhost/private/var/mobile/Library/Mobile%20Documents/498U37S62B~com~ve4d~apple~store~AppNameHere/Data/A8E08CCD-40B5-40F5-A8FA-C336A49D22C3/";
}),
    <PFUbiquityPeer: 0x20bd15e0> (entity: PFUbiquityPeer; id: 0x20bd8050 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityPeer/p2> ; data: {
    localPeerStates = "<relationship fault: 0x20be1510 'localPeerStates'>";
    peerCode = "mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15";
    peerKey = "mobile.50700CD4-20AC-5E8D-BEA9-81F836061E15";
    primaryKeyRanges = "<relationship fault: 0x20be1550 'primaryKeyRanges'>";
    remoteStates = "<relationship fault: 0x20be1590 'remoteStates'>";
    transactionEntries =     (
        "0x20bd4e20 <x-coredata://6D9F3C82-BA7D-4687-BD52-2E2468424B4C/PFUbiquityTransactionEntry/p34>",
        "0x20ba4b30 <x-coredata:///PFUbiquityTransactionEntry/tF1CBC536-F13D-4EAE-8126-DE2C2518977E11>",
        "0x20a7d6f0 <x-coredata:///PFUbiquityTransactionEntry/tF1CBC536-F13D-4EAE-8126-DE2C2518977E12>"
    );
})
)}
Deleted: {(
)}
```

iDevice: Reboot Application:
Quit app and reopen, the records appear as expected. Have to fully quit and reopen.

Rerunning same process on the previous build:

iDevice: Launch with iCloud Enabled:
2013-05-09 00:35:29:009 AppNameHere[8406:2311] UbiquityStoreManager: (Re)loading store...
2013-05-09 00:35:29:017 AppNameHere[8406:17711] UbiquityStoreManager: Will load cloud store: A8E08CCD-40B5-40F5-A8FA-C336A49D22C3 (definite).
2013-05-09 00:35:29:031 AppNameHere[8406:17711] UbiquityStoreManager: Clearing stores...
2013-05-09 00:35:29:036 AppNameHere[8406:17711] willLoadStoreIsCloud: MOC - (null)
2013-05-09 00:35:29:039 AppNameHere[8406:17711] willLoadStoreIsCloud: PSC - (null)

2013-05-09 00:35:29:835 AppNameHere[8406:17711] UbiquityStoreManager: Cloud enabled and successfully loaded cloud store.
2013-05-09 00:35:29:881 AppNameHere[8406:2311] didLoadStoreForCoordinator: MOC (OLD) - (null)
2013-05-09 00:35:29:892 AppNameHere[8406:2311] didLoadStoreForCoordinator: PSC (OLD) - (null)
2013-05-09 00:35:29:898 AppNameHere[8406:2311] didLoadStoreForCoordinator: MOC - (NEW) <NSManagedObjectContext: 0x1f5adf00>
2013-05-09 00:35:29:902 AppNameHere[8406:2311] didLoadStoreForCoordinator: PSC - (NEW) <NSPersistentStoreCoordinator: 0x2009f8b0>

iDevice: Data sent from a second device:

Wrote pending entries to disk after imports finished.
2013-05-09 00:41:16:597 AppNameHere[8406:37915] managedObjectContextForUbiquityChangesInManager: MOC - <NSManagedObjectContext: 0x1f5adf00>
2013-05-09 00:41:16:602 AppNameHere[8406:37915] managedObjectContextForUbiquityChangesInManager: PSC - <NSPersistentStoreCoordinator: 0x2009f8b0>
2013-05-09 00:41:16:606 AppNameHere[8406:37915] UbiquityStoreManager: Importing ubiquity changes into application's MOC. Changes:
{
    deleted = "{(\n)}";
    inserted = "{(\n)}";
    updated = "{(\n    0x1f5f1c10 x-coredata://46DFFF0D-38FD-4F4C-B7A1-0B586696744D/DeviceDetails/p2\n)}";
}

The Table reloads showing the new data with updates as expected.

**Comment by lhunath** [comment]
It looks like USM may not be properly re-registering for notifications of changes when the PSC gets reset.  I'll look into it.

**Comment by lhunath** [comment]
Pushed a fix in 4c0d1af819cf0c667c6c3e1cb094e37b1f6cbc8e - kindly confirm.

**Comment by Tw0bit** [comment]
Just updated our codebase, the notifications continue to be pushed - so appears fixed.
